packages/plugins/visitor-plugin-common/src/base-resolvers-visitor.ts

```diff
@@ -217,7 +217,7 @@ export class BaseResolversVisitor<
 
     const block = new DeclarationBlock(this._declarationBlockConfig)
       .export()
-      .asKind('interface')
+      .asKind('type')
```

Comment:
Are you sure? I think `class` can't extend/implement a `type`.

Reply:
It can, I guess when using `type A = { ... }` TS reads it as `type A = anonymous interface {...}`

https://stackblitz.com/edit/typescript-j8xye4?file=index.ts